src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -211,8 +234,8 @@ private function evaluateBracket(string $expr, mixed $value): array
         }
 
         // filter expressions
-        if (preg_match('/^\?(.*)$/', $expr, $matches)) {
-            $filterExpr = $matches[1];
+        if (preg_match('/^\?\s*(.*)$/', $expr, $matches)) {
```

Comment:
```suggestion
        if (preg_match('/^\?\s*+(.*)$/', $expr, $matches)) {
```

In this case, we want the possessive quantifier to ensure that all spaces are always consumed by the `\s*` part, as we have 2 consecutive quantifiers that could consume them, which is the typical ReDoS pattern.
Alternatively, let spaces be consumed by the `(.*)` (as done before) and be cleaned by the trimming on next line.